api/src/pcapi/core/bookings/api.py

```diff
@@ -540,13 +544,49 @@ def get_individual_bookings_from_stock(stock_id: int) -> typing.Generator[Bookin
         yield booking
 
 
-def archive_old_activation_code_bookings() -> None:
-    old_bookings = Booking.query.join(Booking.stock, Stock.offer, Booking.activationCode).filter(
-        offers_models.Offer.isDigital.is_(True),  # type: ignore [attr-defined]
-        offers_models.ActivationCode.id.isnot(None),
-        Booking.dateCreated < datetime.datetime.utcnow() - constants.ARCHIVE_DELAY,
+def archive_old_bookings() -> None:
+    date_condition = Booking.dateCreated < datetime.datetime.utcnow() - constants.ARCHIVE_DELAY
+    dateUsed = datetime.datetime.utcnow()
+
+    query_old_booking_ids = (
+        Booking.query.join(Booking.stock, Stock.offer)
+        .filter(date_condition)
+        .filter(
+            offers_models.Offer.subcategoryId.in_([c.id for c in FREE_OFFER_CATEGORIES_TO_ARCHIVE]),
+            offers_models.Stock.price == 0,
+        )
+        .options(sa_orm.load_only(Booking.id))
+        .with_entities(Booking.id)
+    )
+
+    number_updated = Booking.query.filter(Booking.id.in_(query_old_booking_ids)).update(
+        {"displayAsEnded": True, "status": BookingStatus.USED, "dateUsed": dateUsed},
+        synchronize_session=False,
     )
-    number_updated = Booking.query.filter(Booking.id.in_(old_bookings.with_entities(Booking.id))).update(
+    db.session.commit()
+
+    logger.info(
+        "Old bookings archived (displayAsEnded=True, status='USED', dateUsed=%s)",
+        dateUsed.strftime("%d/%m/%Y"),
+        extra={
+            "archivedBookings": number_updated,
+        },
+    )
+
+    query_old_digital_booking_ids = (
+        Booking.query.join(Booking.stock, Stock.offer, Booking.activationCode)
+        .filter(date_condition)
+        .filter(
+            offers_models.Offer.isDigital.is_(True),  # type: ignore [attr-defined]
+            offers_models.ActivationCode.id.isnot(None),
+        )
+        .options(sa_orm.load_only(Booking.id))
+        .with_entities(Booking.id)
+    )
+
+    number_updated = Booking.query.filter(
+        Booking.id.in_(query_old_digital_booking_ids.with_entities(Booking.id))
```

Comment:
```suggestion
        Booking.id.in_(query_old_digital_booking_ids)
```
car tu as ajouté `.with_entities(Booking.id)` au-dessus